Framework/PythonInterface/plugins/algorithms/LoadLamp.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import (absolute_import, division, print_function)
+from mantid.api import FileProperty, WorkspaceProperty, PythonAlgorithm, NumericAxis, AlgorithmFactory, FileAction
+from mantid.kernel import Direction
+from mantid.simpleapi import CreateWorkspace, AddSampleLogMultiple, mtd
+
+import numpy
+import h5py
+
+
+class LoadLamp(PythonAlgorithm):
+    def name(self):
+        return 'LoadLamp'
+
+    def category(self):
+        return 'DataHandling\\Nexus'
+
+    def summary(self):
+        return 'Loads HDF files exported from LAMP program at the ILL'
+
+    def PyInit(self):
+        self.declareProperty(FileProperty(name="InputFile", defaultValue="", action=FileAction.Load, extensions=["hdf"]))
+        self.declareProperty(WorkspaceProperty(name="OutputWorkspace", defaultValue="", direction=Direction.Output))
+
+    def PyExec(self):
+        input_file = self.getProperty("InputFile").value
+        output_ws = self.getPropertyValue("OutputWorkspace")
+
+        with h5py.File(input_file, 'r') as hf:
+            DATA = numpy.array(hf.get('entry1/data1/DATA'), dtype='float')
+            if len(DATA.shape) > 2:
```

Comment:
It's not immediately obvious what `len(foo.shape)` does - could this be put in a helper such as `def num_dimensions(array): return len(array.shape)` to make this slightly clearer?

Reply:
I should have used `ndim` directly.